lib/install/management/create.go

```diff
@@ -53,10 +53,12 @@ func (d *Dispatcher) CreateVCH(conf *config.VirtualContainerHostConfigSpec, sett
 	}
 
 	if err = d.createBridgeNetwork(conf); err != nil {
+		d.cleanupAfterCreationFailed(conf, false)
 		return err
 	}
 
 	if err = d.createAppliance(conf, settings); err != nil {
+		d.cleanupAfterCreationFailed(conf, true)
```

Comment:
Can/should this be registered as a defer which checks for success to ensure it's run in all cases? (It seems like there are disadvantages to this approach. I'm not sure what the right answer is.)

Reply:
I'm not very sure about that. The cleanup step only applies to two creation steps between pool creation and appliance VM creation. For the steps before creating the pool, there's nothing to cleanup; for the steps after the appliance VM is created, it's more involved since the VM is created, if it fails anywhere during these steps the VM might be broken as well.